app/controllers/Mod.scala

```diff
@@ -290,6 +291,19 @@ final class Mod(
   def communicationPublic(username: UserStr)  = communications(username, priv = false)
   def communicationPrivate(username: UserStr) = communications(username, priv = true)
 
+  def fullCommsExport(username: UserStr) =
+    SecureBody(_.FullCommsExport) { ctx ?=> me ?=>
+      Found(env.user.repo.byId(username)): user =>
+        val source = env.msg.api
+          .modFullCommsExport(user.id)
+          .map: (tid, msgs) =>
+            s"=== 0 === thread: ${tid}\n${msgs.map(m => s"${m.date} ${m.user}: ${m.text}\n--- 0 ---\n").toList.mkString("\n")}"
```

Comment:
The separators are a bit strange, but were chosen to be sure they wouldn't be found in the message content, and to quickly go from thread to thread.